networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,301 @@
+
+
+    Returns
+    -------
+    nodes : dictionary
+        Dictionary keyed by node with DomiRank centrality of the node as value.
+
+    sigma : float
+        $\alpha$ normalized by the smallest eigenvalue.
+
+    converged: boolean
+        Whether the centrality computation converged. Returns ``None`` if ``analytical = True``.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> centrality, sigma, converged = nx.domirank(G)
+    >>> print([f"{node} {centrality[node]:0.2f}" for node in centrality])
+    ['0 -0.54', '1 1.98', '2 -1.08', '3 1.98', '4 -0.54']
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    NetworkXUnfeasible
+        If alpha is negative (and thus outside its bounds): ``alpha < 0``.
+
+    NetworkXUnfeasible
+        If ``alpha > 1`` with the ``analytical = False`` argument.
+
+    NetworkXAlgorithmError
+        If ``patience > max_iter``.
+
+    NetworkXAlgorithmError
+        If ``patience < 5``.
+
+    NetworkXAlgorithmError
+        If dt is not in the bounds ``0 < dt < 1``.
+
+    NetworkXAlgorithmError
+        If epsilon is negative ``epsilon < 0``.
+
+    Warning
+        If supercharging the competition parameter for the analytical solution: ``alpha > 1``.
+
+    Warning
+        If one is using the analytical solution for a large system, i.e. ``len(G) > 5000``, as the algorithm will be slow.
+
+    See Also
+    --------
+    :func:`~networkx.algorithms.centrality.degree_centrality`
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
+        Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0
+    """
+    import numpy as np
+    import scipy as sp
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph."
+        )
+    if patience > max_iter:
+        raise nx.NetworkXAlgorithmError("it is mandatory that `max_iter > patience`.")
+    if patience < 5:
+        raise nx.NetworkXAlgorithmError("it is mandatory that `patience >= 5`.")
+    if dt < 0 or dt > 1:
+        raise nx.NetworkXAlgorithmError(
+            "it is mandatory that dt is bounded such that: `0 < dt <= 1`."
+        )
+    if epsilon < 0:
+        raise nx.NetworkXAlgorithmError(
+            "it is mandatory that `epsilon > 0` and recommended that `epsilon = 1e-5`."
+        )
+    GAdj = nx.to_scipy_sparse_array(G)  # convert to scipy sparse csr array
+
+    # Here we create a warning (I couldn't find a networkxwarning, only exceptions and erros), that suggests to use the iterative formulation of DomiRank rather than the analytical form.
+    if GAdj.shape[0] > 5000 and analytical == True:
```

Comment:
If you don't replace `analytical` with `method`, this should be `... and analytical` for consistency.